openmls/src/group/core_group/apply_proposals.rs

```diff
@@ -57,6 +57,35 @@ impl CoreGroup {
         let mut self_removed = false;
         let mut external_init_secret_option = None;
 
+        // Process external init proposals. We do this before the removes, so we
+        // know that removing "ourselves" (i.e. removing the group member in the
+        // same leaf as we are in) is valid in this case. We only care about the
+        // first proposal and ignore all others.
+        if let Some(queued_proposal) = proposal_queue
+            .filtered_by_type(ProposalType::ExternalInit)
+            .next()
+        {
+            let external_init_proposal = &queued_proposal
+                .proposal()
+                .as_external_init()
+                // We know the proposal type, so it's a library error if we're wrong.
+                .ok_or(CoreGroupError::LibraryError)?;
```

Comment:
This could be replaced with an `if let` without an `else` clause to avoid the library error.